packages/catalog-process/src/services/catalogService.ts

```diff
@@ -403,43 +403,91 @@ export function catalogServiceBuilder(
     },
 
     async createEService(
-      apiEServicesSeed: catalogApi.EServiceSeed,
+      seed: catalogApi.EServiceSeed,
       { authData, correlationId, logger }: WithLogger<AppContext>
     ): Promise<EService> {
-      logger.info(
-        `Creating EService with service name ${apiEServicesSeed.name}`
-      );
+      logger.info(`Creating EService with name ${seed.name}`);
 
       if (!config.producerAllowedOrigins.includes(authData.externalId.origin)) {
         throw originNotCompliant(authData.externalId.origin);
       }
 
       const eserviceWithSameName =
         await readModelService.getEServiceByNameAndProducerId({
-          name: apiEServicesSeed.name,
+          name: seed.name,
           producerId: authData.organizationId,
         });
       if (eserviceWithSameName) {
-        throw eServiceDuplicate(apiEServicesSeed.name);
+        throw eServiceDuplicate(seed.name);
       }
 
+      const creationDate = new Date();
       const newEService: EService = {
         id: generateId(),
         producerId: authData.organizationId,
-        name: apiEServicesSeed.name,
-        description: apiEServicesSeed.description,
-        technology: apiTechnologyToTechnology(apiEServicesSeed.technology),
-        mode: apiEServiceModeToEServiceMode(apiEServicesSeed.mode),
+        name: seed.name,
+        description: seed.description,
+        technology: apiTechnologyToTechnology(seed.technology),
+        mode: apiEServiceModeToEServiceMode(seed.mode),
         attributes: undefined,
         descriptors: [],
-        createdAt: new Date(),
+        createdAt: creationDate,
         riskAnalysis: [],
       };
 
-      const event = toCreateEventEServiceAdded(newEService, correlationId);
-      await repository.createEvent(event);
+      const eserviceCreationEvent = toCreateEventEServiceAdded(
+        newEService,
+        correlationId
+      );
+
+      if (
+        seed.descriptor.dailyCallsPerConsumer > seed.descriptor.dailyCallsTotal
+      ) {
+        throw inconsistentDailyCalls();
+      }
+
+      const draftDescriptor: Descriptor = {
+        id: generateId(),
+        description: seed.descriptor.description,
+        version: "1",
+        interface: undefined,
+        docs: [],
+        state: descriptorState.draft,
+        voucherLifespan: seed.descriptor.voucherLifespan,
+        audience: seed.descriptor.audience,
+        dailyCallsPerConsumer: seed.descriptor.dailyCallsPerConsumer,
+        dailyCallsTotal: seed.descriptor.dailyCallsTotal,
+        agreementApprovalPolicy:
+          apiAgreementApprovalPolicyToAgreementApprovalPolicy(
+            seed.descriptor.agreementApprovalPolicy
+          ),
+        serverUrls: [],
+        publishedAt: undefined,
+        suspendedAt: undefined,
+        deprecatedAt: undefined,
+        archivedAt: undefined,
+        createdAt: creationDate,
+        attributes: { certified: [], declared: [], verified: [] },
+      };
```

Comment:
```suggestion
      const draftDescriptor: Descriptor = {
        ...seed.descriptor,
        id: generateId(),
        state: descriptorState.draft,
        version: "1",
        agreementApprovalPolicy:
          apiAgreementApprovalPolicyToAgreementApprovalPolicy(
            seed.descriptor.agreementApprovalPolicy
          ),
        serverUrls: [],
        createdAt: creationDate,
        attributes: { certified: [], declared: [], verified: [] },
        docs: [],
      };
```
We can make this definition clearer by using spreading and omit the undefined values

Reply:
It's better to avoid spreading the seed directly, so we have more control on what is being passed.